docs/develop/cmodules.rst

```diff
@@ -117,9 +117,9 @@ Directory::
     # This is not actually needed in this example.
     CFLAGS_USERMOD += -I$(EXAMPLE_MOD_DIR)
 
-Finally you will need to modify the ``mpconfigboard.h`` for your board
-to tell the build process to include the new module by adding the
-following
+Finally you will need to modify the ``mpconfigport.h`` (found under
+micropython/ports/PORTNAME) for your port to tell the build process
+to include the new module by adding the following
```

Comment:
I think this is the wrong advice to give here.  The idea with the C modules is that you don't need to touch any code in the port directory, you can do all the config outside it, in your own custom board source files.  So for stm32 it's correct to use `mpconfigboard.h`.  Other ports don't have this level of configurability.  For such ports it might be better to use a make variable on the command line, like `make CFLAGS_EXTRA=-DMODULE_EXAMPLE_ENABLED=1`  The problem with that is that not all ports have `CFLAGS_EXTRA`.

So I'd suggest keeping the `mpconfigboard.h` wording but adjust it to say it's specifically for stm32 (and I think it'll also work on the nrf port), and then add extra wording to say something like: "or manually edit `mpconfigport.h` if the port doesn't support `mpconfigboard.h" files.